tests/dummy/app/templates/docs/faq/routing-and-asset-loading.md

```diff
@@ -0,0 +1,101 @@
+
+This request intercepting code extracts the requested URL path, appends it to the path to the root of the Ember app, and then looks to see if the result is the path to a file inside the Ember app. If so, it resolves the request to that file. If not, it resolves it to the full requested (absolute) path.
+
+This is not perfect -- there is a concievable case where it could do the wrong thing, which is if the URL path pointed to a file when interpreted either as absolute or relative to the Ember app. For example, if you put `images/foo.jpg` in your Ember app's `public` directory, then a request for `/images/foo.jpg` would resolve to it. If for some reason the application were trying to load `/images/foo.jpg` from the root of the filesystem (i.e. you have a top-level directory called `images` and the app expects it to contain `foo.jpg`), it would resolve to the one inside the Ember app, and the application would have no way to load the one at the root of the filesystem. This seems like a very unlikely use case, though -- applications that load files from the filesystem generally do it through node APIs, not in-browser requests.
+
+### Source maps
+
+One might observe that the solution for images, fonts, etc., could also apply to the Javascript and CSS, and wonder why we need both solutions. The answer is source maps. Unfortunately when the Chromium developer tools load source maps, it is not intercept-able by the Electron `protocol` module, and the developer tools are not aware of any intercepting that happens in the main process. So if we left the leading `/` on the script tags, e.g. `<script src="/assets/vendor.js"></script>`, then when the developer tools sees the `sourceMappingURL=vendor.map` at the end of `vendor.js` it will treat `vendor.map` as a path that is relative to where it thinks the script was loaded from, i.e. `/assets/vendor.js`, so it will try to load it from `/assets/vendor.map`. Since this request isn't intercept-able, the source maps will fail to load. So to make source maps work, we really need to load the Javascript and CSS files from relative URLs, and since we cannot easily rewrite the other assets' URLs to be relative, we need two different solutions for the two different flavors of assets.
```

Comment:
👍 This answered the question I had from the beginning of this FAQ article. Thank you for including these details.